README.md

```diff
@@ -1,126 +1,99 @@
 # Balancer rewards manager
 
-This repository contains Lido rewards controller for [Balancer Merkle Orchard contract](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/distributors/contracts/MerkleOrchard.sol) and manager for simplifying managing it via DAO voting.
-Rewarder weekly approves certain amount of LDO to be spendable by Balancer contract. And provides interface to general rewards manager.
+This repository contains rewards manager for [Balancer Liquidity Gauge](https://etherscan.io/address/0xcD4722B7c24C29e0413BDCd9e51404B4539D14aE#code). The manager simplifies reward distribution operations by DAO voting and Easy Track.
 
-## Deploying Environment
-
-`deploy.py` script that deploys `BalancerRewardsController` and `RewardsManager` contracts. The script next ENV variables be set.
+## Environment preparation with Poetry
 
-`DEPLOYER` deployer account
-
-## BalancerRewardsController.vy
-### Balancer side
+Step 1. Install dependencies
+```shell
+poetry install
+```
 
-##### `view available_allowance() -> uint256`
+Step 2. Activate newly created environment
+```shell
+poetry shell
+```
 
-Returns current allowance limit available for distribution by calling `createDistribution`
+## Testing
 
-##### `createDistribution(_token: address, _merkle_root: bytes32, _amount: uint256, _distribution_id: uint256):`
+`brownie test -s`
 
-Wrapper for `createDistribution` of MerkleOrchard contract.
+## Deploying Environment
 
-Wrapper for `createDistribution(token: ERC20, merkleRoot: bytes32, amount: uint256, distributionId: uint256)`
-of Merkle Orchard contract and allowes to distibute LDO token holded by this contract
-with amount limited by available_allowance()
+The `deploy.py` script is in charge of the `RewardsManager` contract on-chain deployment.
+The following environment variables needs to be set for the script's execution:
 
-Reverts if `_amount` is greater than Manager balance or allowance limit.
-Reverts if `_token` is not LDO.
-Reverts if contract is paused.
-Reverts if contract has not enough balance to distribute `_amount` of LDO
+* `DEPLOYER` - deployer account
 
-Events:
+## Specification
 
-```vyper=
-event RewardsDistributed:
-    amount: uint256
-```
+#### RewardsManager.sol
 
-##### `set_balancer_distributor(_new_balancer_distributor: address)`
+**def period_finish() -> uint256: view**
 
-Changes `BALANCER_DISTRIBUTOR`. Can be called by owner or current balancer_distributor.
+Returns estimated date of last rewards period start date
 
-Events:
+    BLG.periodFinish + (LDO.balanceOf(self)/self.rewardAmount) * SECONDS_PER_WEEK
```

Comment:
The formula differs from the implementation provided here: https://github.com/lidofinance/balancer-rewards-manager/blob/feature/v3/contracts/RewardsManager.vy#L148:L159

Also, the doc comment also has a slightly different meaning.